torch/csrc/autograd/engine.cpp

```diff
@@ -335,26 +360,30 @@ auto Engine::thread_main(
       // 'mark_graph_task_completed' would mark the Future as completed and this
       // would notify the owner thread that the task has been completed.
       mark_graph_task_completed(local_graph_task);
+
+      // If the graph_task marked as completed, then the owning thread should
+      // finish and exit, owning thread don't need to wait for future tasks.
```

Comment:
Maybe something like, "The CPU worker thread is actually the thread that initially requested the autograd computation. Now that we're done, break out of the worker loop so we can continue executing the rest of the calling code!"